### PR TITLE
Convert CharSpan to C++ String for proper comparison.

### LINKED ISSUE #19475: When entering 'y' in UserPromt, the comparison is not successful.
#### Problem
When entering y in the UserPrompt the comparison is not successful, because it was converted to CharSpan and what is being compared is the entire buffer in data(), when it contains garbage on purpose. 

 #### Proposed Solution
Convert the CharSpan to a C++ string using the correct size and do the comparison.

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
When comparing CharSpan to C++ String, we need to convert CharSpan to a string and then do the comparison. 
* Fix incorrect comparison in UserPrompt.
* Fixes #19475 

#### Change overview
- Convert CharSpan to C++ string using the correct size and then do the comparison. 

#### Testing
* Validated that the input now correctly accepts the value given.

### CODE REVIEW

File: src/app/tests/suites/commands/log/LogCommands.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ CHIP_ERROR LogCommands::UserPrompt(const char * identity,
     {
         std::string line;
         std::getline(std::cin, line);
-        if (line != value.expectedValue.Value().data())
+        if (line.compare(std::string(value.expectedValue.Value().data(), value.expectedValue.Value().size())))
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/commands/log/LogCommands.cpp:37]: Can't this still use `!=`?  Why the change to `compare`?

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
Fast-tracking test tool change.

File: src/app/tests/suites/commands/log/LogCommands.cpp (hunk 1 above)
Comment (krypton36) [reply src/app/tests/suites/commands/log/LogCommands.cpp]: Updated.

**Review by andy31415 — APPROVED** [review]

**Review by harsha-rajendran — APPROVED** [review]
